app/models/journal/aggregated_journal.rb

```diff
@@ -41,6 +41,12 @@
 #    be dropped
 class Journal::AggregatedJournal
   class << self
+    # Returns the aggregated journal that contains the specified (vanilla) journal.
+    def for_journal(vanilla_journal)
+      Journal::AggregatedJournal.aggregated_journals(journable: vanilla_journal.journable)
+                  .detect { |journal| journal.version >= vanilla_journal.version }
```

Comment:
By relying on 'detect', you are fetching all the journals from the DB, instantiate them in ruby, and then filter for only one of them.
You could instead use a where clause to filter in the db.